lib/PlacesProvider.js

```diff
@@ -668,6 +669,88 @@ Links.prototype = {
   }),
 
   /**
+   * Gets the remote tabs links.
+   *
+   * @param {Integer} options
+   *
+   *
+   * @returns {Promise} Returns a promise with the array of links as payload.
+   */
+  getRemoteTabsLinks:  Task.async(function*(options = {}) {
+    let {limit} = options;
+    if (!limit || limit.options > LINKS_QUERY_LIMIT) {
+      limit = LINKS_QUERY_LIMIT;
+    }
+
+    // We first get the URLs open on other devices from SyncedTabs.
+    let clients = yield SyncedTabs.getTabClients();
+
+    if (clients.length === 0) {
+      // No reason to waste cycles doing anything else.
+      return;
+    }
+
+    // get the URLs with the data we need from all the clients
+    let urls = {};
+    for (let client of clients) {
+      for (let tab of client.tabs) {
+        urls[tab.url] = {
+          deviceName: client.name,
+          deviceIsMobile: client.isMobile,
+          lastUsed: tab.lastUsed
+        };
+      }
+    }
+
+    // Now we add the data required from the places DB.
+    // setup binding parameters
+    let params = {
+      limit: limit
+    };
+
+    let blockedURLs = this.blockedURLs.items().map(item => `"${item}"`);
+
+    let urlsToSelect = Object.keys(urls).map(item => `"${item}"`); //
+
+    // construct sql query
+    let sqlQuery =  `SELECT moz_places.url as url,
+                            moz_places.guid as guid,
+                            moz_favicons.data as favicon,
+                            moz_favicons.mime_type as mimeType,
+                            moz_places.title,
+                            moz_places.frecency,
+                            moz_places.last_visit_date / 1000 as lastVisitDate,
+                            "history"  as type,
```

Comment:
@rlr can we use a more specific `type` for the synced/remote history links here?
According to #935, we want to track highlight source in the click ping.